src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +165,72 @@ public boolean equals(Object o) {
         return true;
     }
 
+    /** return a MD5 sum for ths dictionary, the checksum is not re-computed each time this
```

Comment:
I don't see how it is not re-computed each time.  In the case that it doesn't re-compute, it perhaps should when new sequences are set.